types/index.d.ts

```diff
@@ -434,7 +434,8 @@ export type Admin = {
   fetchTopicMetadata(options?: { topics: string[] }): Promise<{ topics: Array<ITopicMetadata> }>
   fetchOffsets(options: {
     groupId: string
-    topic: string
+    topic?: string
```

Comment:
Github won't let me create a suggestion that goes outside of the lines of the diff, but there's an issue here where the return type of this function is different depending on the input, so we need to be a little more clever here. There's probably some fancy-pants thing you can do with conditional types, but we can also just use plain old method overloading:

```ts
interface FetchOffsetsBaseOptions {
  groupId: string,
  resolveOffsets?: boolean
}

export interface FetchOffsetsTopicOptions extends FetchOffsetsBaseOptions {
  topic: string
}

export interface FetchOffsetsTopicsOptions extends FetchOffsetsBaseOptions {
  topics: string[]
}

export type Admin = {
  fetchOffsets(options: FetchOffsetsTopicOptions): Promise<Array<PartitionOffset & { metadata: string | null }>>,
  fetchOffsets(options: FetchOffsetsTopicsOptions): Promise<Array<{ topic: string, partitions: Array<PartitionOffset & { metadata: string | null }>}>>,
}
```

This has a few benefits. The first is that the type system will prevent `admin.fetchOffsets({ topic: 'foo', topics: ['bar'] })`, because the `topic` and `topics` properties are mutually exclusive. The second is that it allows us to define the relationship between the input type and the return type. The return type is a bit of an abomination. If we didn't return `metadata` it could be simplified to just `Promise<TopicOffsets[]>` and `Promise<PartitionOffset[]>`, but alas, here we are.

Reply:
Done. I've opted to keep the options parameter inline though, but moved the `Array<PartitionOffset & { metadata: string | null }>` to a new type. LMK if you prefer the options to be extracted to a type as well.